ui/src/app/lib/atlasmap-data-mapper/services/mapping-management.service.ts

```diff
@@ -187,20 +187,27 @@ export class MappingManagementService {
    * @param field
    */
   transitionMode(mapping: MappingModel, field: Field): void {
+    this.transitionMode2(mapping, field.isSource());
     if (mapping.transition.mode === TransitionMode.ONE_TO_ONE) {
-      const mappedFields: MappedField[] = mapping.getMappedFields(field.isSource());
-      if (mappedFields.length > 2) {
-        if (field.isSource()) {
+      this.transitionMode2(mapping, !field.isSource());
```

Comment:
Why did you create a transitionMode2 out of transitionMode()??  These should remain as transitionMode().

Reply:
To deal with arrays and collections, I needed to analyze both the source and target fields in the mapping, so needed to call the method for both sides.

Reply:
transitionMode is really only trying to move from map-mode to either many to one or one to many.  Field.isSource() tells you which way - I don't see why calling it twice to run on all of the sources and targets is needed. Not clear how this was necessary for your issue.

Reply:
I also think this is unrelated to #905, please remove this change.